include/mptcpd/types.h

```diff
@@ -103,31 +103,59 @@ struct mptcpd_limit
- * the result of calling @c mptcpd_pm_get_addr() or
- * @c mptcpd_pm_dump_addrs() is available.
+ * the result of calling @c mptcpd_kpm_get_addr() or
+ * @c mptcpd_kpm_dump_addrs() is available.
  *
  * @param[in]     info          Network address information.  @c NULL
  *                              on error.
  * @param[in,out] callback_data Data provided by the caller of
- *                              @c mptcpd_pm_get_addr() or
- *                              @c mptcpd_pm_dump_addrs().
+ *                              @c mptcpd_kpm_get_addr() or
+ *                              @c mptcpd_kpm_dump_addrs().
  */
-typedef void (*mptcpd_pm_get_addr_cb)(struct mptcpd_addr_info const *info,
-                                      void *callback_data);
+typedef void (*mptcpd_kpm_get_addr_cb_t)(
+        struct mptcpd_addr_info const *info,
+        void *callback_data);
+
+/**
+ * @brief Type of function called on asynchronous call completion.
+ *
+ * The mptcpd path manager API has several functions that complete
+ * asynchronously.  Those functions accept a parameter of this type to
+ * allow the caller to be notified of completion of the asynchronous
+ * functions.
+ *
+ * Functions of this type differ from user provided callbacks that are
+ * called when asynchronous results are available in that this type of
+ * function is called regardless of whether or not asynchronous
+ * results are available.  Furthermore, they are only called once at
+ * the very end of the asynchronous call, whereas those called upon
+ * availability of results may be called multiple times for a single
+ * asynchronous call, such as the @c mptcpd_kpm_dump_addrs() case.
+ *
+ * Functions of this type are suitable for deallocating dynamically
+ * allocated user_data passed to asynchronous calls, for example.
+ *
+ * @param [in,out] user_data Data provided the user at the time of the
+ *                           asynchronous call.
+ */
+typedef void (*mptcpd_complete_func_t)(void *user_data);
+
+/// Type of function called on async path manager call completion.
+typedef mptcpd_complete_func_t mptcpd_kpm_complete_func_t;
 
 /**
  * @brief Type of function called when MPTCP resource limits are
  *        available.
  *
  * The mptcpd path manager will call a function of this type when
- * the result of calling @c mptcpd_pm_get_limits() is available.
+ * the result of calling @c mptcpd_kpm_get_limits() is available.
  *
  * @param[in]     limits        Array of MPTCP resource type/limit
  *                              pairs.  @c NULL on error.
  * @param[in]     len           Length of the @a limits array.  Zero
  *                              on error.
  * @param[in,out] callback_data Data provided by the caller of
- *                              @c mptcpd_pm_get_limits().
+ *                              @c mptcpd_kpm_get_limits().
  */
 typedef void (*mptcpd_pm_get_limits_cb)(
         struct mptcpd_limit const *limits,
```

Comment:
I'll probably rename this typedef to `mptcpd_kpm_get_limits_cb_t` in another pull request for the sake of consistency.